HudHud/Navigation/Engine/Replanner/BetterRouteDecider.swift

```diff
@@ -0,0 +1,231 @@
+        // Street should be at least 500m of the new route
+        // Avoids showing tiny street names
+        // But not so big that we miss important connecting roads
+        let minStreetDistance: Measurement<UnitLength>
+    }
+
+    // MARK: Properties
+
+    private let analyzer: RouteAnalyzer
+    private let config: Config
+
+    // MARK: Lifecycle
+
+    init(analyzer: RouteAnalyzer, config: Config) {
+        self.analyzer = analyzer
+        self.config = config
+    }
+
+    // MARK: Functions
+
+    func evaluateAlternatives(
+        _ alternatives: [Route],
+        comparedTo current: Route,
+        at position: ExactRoutePosition,
+        and progress: TripState
+    ) -> RouteDecision {
+        var bestBetterRoute: BetterRoute?
+
+        for alternative in alternatives {
+            if let betterRoute = analyze(
+                alternative: alternative,
+                against: current,
+                at: position,
+                progress: progress
+            ) {
+                if isBetter(betterRoute, than: bestBetterRoute) {
+                    bestBetterRoute = betterRoute
+                }
+            }
+        }
+
+        return makeFinalDecision(bestBetterRoute)
+    }
+}
+
+private extension BetterRouteDecider {
+
+    func analyze(
+        alternative: Route,
+        against current: Route,
+        at position: ExactRoutePosition,
+        progress: TripState
+    ) -> BetterRoute? {
+        guard let currentProgress = progress.currentProgress else {
+            return nil
+        }
+
+        let comparison = self.analyzer.analyzeAlternative(
+            current: current,
+            alternative: alternative,
+            considering: position,
+            and: currentProgress
+        )
+
+        guard self.isViableAlternative(comparison) else {
+            return nil
+        }
+
+        let forkPoint = self.analyzer.findForkPoint(
+            current: current,
+            alternative: alternative,
+            userPosition: position
+        )
+
+        let betterRoute = BetterRoute(
+            route: alternative,
+            timeSaving: .seconds(comparison.timeDifference),
+            distanceImpact: .meters(comparison.distanceDifference),
+            viaStreet: self.findMostDistinctiveStreet(
+                in: alternative,
+                comparedTo: current,
+                at: progress
+            ),
+            trafficConfidence: self.calculateConfidence(comparison.trafficImpact),
+            forkPoint: forkPoint
+        )
+
+        return betterRoute
+    }
+
+    func isViableAlternative(_ comparison: RouteAnalyzer.RouteComparison) -> Bool {
+        if comparison.timeDifference > self.config.minTimeSaving.seconds {
+            return true
+        }
+
+        let requiredSaving = comparison.remainingPercentage > self.config.progressThreshold ?
+            self.config.minTimeSaving.seconds * 1.5 :
+            self.config.minTimeSaving.seconds
```

Comment:
This is a bit hard to read, can we move this to a local function and use a normal if?

```suggestion
        func calcualteRequiredSaving() -> Double {
            if comparison.remainingPercentage > self.config.progressThreshold {
                return self.config.minTimeSaving.seconds * 1.5
            } else {
                return self.config.minTimeSaving.seconds
            }
        }
        let requiredSaving = calcualteRequiredSaving()
```

Then my second question is, where does the 50% multiplier come from? Should we expose this as a configurable parameter?

Instead of making this a binary decision, we could also explore the idea for a linear mapping?